retrofit-adapters/rxjava2/src/main/java/retrofit2/adapter/rxjava2/RxJava2CallAdapter.java

```diff
@@ -63,7 +63,7 @@
     if (isResult) {
       observable = new ResultObservable<>(responseObservable);
     } else if (isBody) {
-      observable = new BodyObservable<>(responseObservable);
+      observable = new BodyObservable<>(responseObservable, !isCompletable);
```

Comment:
Possibly! So should Observable and Flowale then. 

When my colleague and I were looking at this, our mental model was basically that if you have an endpoint that tries to coerce everything through onSuccess as a sealed type of some kind, it would be weird if a couple types of status codes basically didn't route that way and would be confusing for consumers that otherwise would never check onComplete. With Completable at least it explicitly doesn't allow for that ambiguity. We could err on the side of flexibility with this though and leave it up to the developer.

Reply:
Thinking on this more, I think this could work if we wanted to make it entirely silent and route through onComplete if possible _only_ if we're fine with not exposing the exception. That is to say - `Single` users would get the `NoContentException`, but everyone else would basically have to use some sort of continuation (`switchIfEmpty()`, `defaultIfEmpty()`, etc) if they need to handle it otherwise. They wouldn't have access to the underlying response, but maybe we make the call that that's ok? It's kind of sneaky/silent behavior, but I also suspect that'd be the common case and not making people always manually handle the `NoContentException` in the common case seems possibly better. Also makes it more consistent, rather than making `Completable` the exception even though others have `onComplete`.

Thoughts?

Reply:
I demoed this in 86f387d, but can revert if we don't want that behavior